Mathlib/FieldTheory/IsSepClosed.lean

```diff
@@ -170,15 +177,108 @@ class IsSepClosure [Algebra k K] : Prop where
   sep_closed : IsSepClosed K
   separable : IsSeparable k K
 
+/-- A separably closed field is its separable closure. -/
+instance IsSepClosure.self_of_isSepClosed [IsSepClosed k] : IsSepClosure k k :=
+  ⟨by assumption, isSeparable_self k⟩
+
 variable {k} {K}
 
 theorem isSepClosure_iff [Algebra k K] :
     IsSepClosure k K ↔ IsSepClosed K ∧ IsSeparable k K :=
   ⟨fun h => ⟨h.1, h.2⟩, fun h => ⟨h.1, h.2⟩⟩
 
-instance (priority := 100) IsSepClosure.normal [Algebra k K]
-    [IsSepClosure k K] : Normal k K :=
-  ⟨fun x => by apply IsIntegral.isAlgebraic; exact IsSepClosure.separable.isIntegral' x,
-    fun x => @IsSepClosed.splits_codomain _ _ _ _ (IsSepClosure.sep_closed k) _ _ (by
-      have : IsSeparable k K := IsSepClosure.separable
-      exact IsSeparable.separable k x)⟩
+-- TODO: move to suitable file
+instance IsSeparable.isAlgebraic [Algebra k K] [IsSeparable k K] : Algebra.IsAlgebraic k K :=
+  fun x => IsIntegral.isAlgebraic k <| IsSeparable.isIntegral' x
+
+namespace IsSepClosure
+
+instance isAlgebraic [Algebra k K] [IsSepClosure k K] : Algebra.IsAlgebraic k K :=
+  have : IsSeparable k K := IsSepClosure.separable
```

Comment:
This could be a global instance, right?

Reply:
Yes, what name do you suggest? `isAlgebraic_of_isSepClosure`?

Reply:
But there are also a lot of instances in mathlib4 declared in their own namespaces...

Reply:
I think Johan is proposing to make `IsSepClosure.separable` an instance, so you don't need the have line.